hazelcast/src/test/java/com/hazelcast/config/ConfigCompatibilityChecker.java

```diff
@@ -223,12 +217,27 @@ private static boolean isCompatible(CollectionConfig c1, CollectionConfig c2) {
                 && isCompatible(c1.getMergePolicyConfig(), c2.getMergePolicyConfig());
     }
 
+    private static boolean isCompatible(MerkleTreeConfig c1, MerkleTreeConfig c2) {
+        boolean c1Disabled = c1 == null || !c1.isEnabled();
+        boolean c2Disabled = c2 == null || !c2.isEnabled();
+        return c1 == c2 || (c1Disabled && c2Disabled) || (c1 != null && c2 != null
+                && nullSafeEqual(c1.getDepth(), c2.getDepth()));
```

Comment:
minor: no reason to use `nullSafeEqual` for `int` comparison here (same below)

Reply:
True, copy-paste error, thanks!